x-pack/plugins/observability_solution/entity_manager/server/lib/entities/ingest_pipeline/generate_history_processors.ts

```diff
@@ -39,7 +39,33 @@ function createMetadataPainlessScript(definition: EntityDefinition) {
     return `${script}if (ctx.entity?.metadata?.${source.replaceAll(
       '.',
       '?.'
-    )} != null) {${mapDestinationToPainless(destination, source)}\n}\n`;
+    )} != null) {${mapMetadataDestinationToPainless(destination, source)}\n}\n`;
+  }, '');
+}
+
+function mapIdentityFieldsDestinationToPainless(identityField: string) {
+  const fieldParts = identityField.split('.');
+  return fieldParts.reduce((acc, _part, currentIndex, parts) => {
+    if (currentIndex + 1 === parts.length) {
+      const target = `ctx${parts.map((s) => `["${s}"]`).join('')}`;
+      const segment = `
+       ${target} = ctx.entity.identityFields.${identityField};
```

Comment:
Since this comes from the composite, the value is just a plain string.

Reply:
is it necessary to have this complex logic vs something more simple like 

```
    ...definition.identityFields.map((key) => {
      return {
        set: {
          field: key.field,
          value: `{{entity.identityFields.${key.field}}}`,
          if: `ctx.entity?.identityFields?.${key.field.replaceAll('.', '?.')} != null`,
        }
      }
    }),
```

does this correctly handle the nested object behaviour?

Reply:
Good question, I'd have to try that

Reply:
Moving to my other PR.